### PR TITLE
feat(condo): DOMA-6054 add BillingReceiptFile model

### PR DESCRIPTION
**Pull request by abshnko**
**New model in Billing domain - BillingReceiptFile**

Fields:

- file: File
- controlSum: Text
- receipt: BillingReceipt (link)
- context: BillingIntegrationOrganizationContext (link)
- importId: Text

UI changes:

Screenshot 2023-05-13 at 22 45 38

### CODE REVIEW

File: apps/condo/domains/billing/access/BillingReceiptFile.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
+// @ts-nocheck
```
Comment (dkoviazin) [inline apps/condo/domains/billing/access/BillingReceiptFile.js:1]: // @ts-nocheck

File: apps/condo/domains/billing/access/BillingReceiptFile.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+ */
+
+const get = require('lodash/get')
+
+const { throwAuthenticationError } = require('@open-condo/keystone/apolloErrorFormatter')
+const { getById } = require('@open-condo/keystone/schema')
+
+const { checkPermissionInUserOrganizationOrRelatedOrganization,
+    queryOrganizationEmployeeFor,
+    queryOrganizationEmployeeFromRelatedOrganizationFor,
```
Comment (dkoviazin) [inline apps/condo/domains/billing/access/BillingReceiptFile.js:13]: please, do not use relative paths, use smth like @condo/domains/....

File: apps/condo/domains/billing/access/BillingReceiptFile.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+} = require('../../organization/utils/accessSchema')
+const { STAFF } = require('../../user/constants/common')
+
+async function canReadBillingReceiptFiles ({ authentication: { item: user } }) {
```
Comment (dkoviazin) [inline apps/condo/domains/billing/access/BillingReceiptFile.js:17]: You need to add access for service users
example is here @condo/domains/billing/access/BillingProperty.js

File: apps/condo/domains/billing/schema/BillingReceiptFile.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+ */
+
+const { Text, Relationship, File } = require('@keystonejs/fields')
+
+const { historical, versioned, uuided, tracked, softDeleted, dvAndSender } = require('@open-condo/keystone/plugins')
+const { GQLListSchema } = require('@open-condo/keystone/schema')
+
+const access = require('@condo/domains/billing/access/BillingReceiptFile')
+const FileAdapter = require('@condo/domains/common/utils/fileAdapter')
+const BILLING_RECEIPT_FILE_FOLDER_NAME = 'billingReceipt'
```
Comment (dkoviazin) [inline apps/condo/domains/billing/schema/BillingReceiptFile.js:12]: s3 will lowercase folder name
better to use 'billing-receipt' or 'billing-receipt-pdf'

File: apps/condo/domains/billing/utils/clientSchema/BillingReceiptFile.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    BillingReceiptFileCreateInput,
+    BillingReceiptFileUpdateInput,
+    QueryAllBillingReceiptFilesArgs,
+} from '@app/condo/schema'
+
+import { generateReactHooks } from '@open-condo/codegen/generate.hooks'
+
+import { BillingReceiptFile as BillingReceiptFileGQL } from '@condo/domains/billing/gql'
+
+// TODO(codegen): write utils like convertToFormState and formValuesProcessor if needed, otherwise delete this TODO
```
Comment (pahaz) [inline apps/condo/domains/billing/utils/clientSchema/BillingReceiptFile.ts:16]: !

File: apps/condo/domains/billing/schema/BillingReceiptFile.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        },
+
+        controlSum: {
+            schemaDoc: 'Meta information about the file',
+            type: Text,
+            isRequired: true,
+        },
+
+    },
+    plugins: [uuided(), versioned(), tracked(), softDeleted(), dvAndSender(), historical()],
```
Comment (pahaz) [inline apps/condo/domains/billing/schema/BillingReceiptFile.js:52]: Do you need to add an organisation field?

File: apps/condo/domains/billing/schema/BillingReceiptFile.test.js [hunk 7]
```diff
@@ -0,0 +1,254 @@
...
+                expect(objs.length).toBeGreaterThanOrEqual(1)
+                expect(objs[0]).toMatchObject({
+                    id: file.id,
+                    context: file.context,
+                    receipt: file.receipt,
+                    controlSum: file.controlSum,
+                })
+            })
+
+            test('anonymous can\'t', async () => {
```
Comment (dkoviazin) [inline apps/condo/domains/billing/schema/BillingReceiptFile.test.js:227]: You need to add 2 cases employee with the same organization - can read
and employee from another organization - can't

File: apps/condo/domains/billing/access/BillingReceiptFile.js [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+
+async function canReadBillingReceiptFiles ({ authentication }) {
+    const user = authentication.item
+    if (!user) return throwAuthenticationError()
+    if (user.deletedAt) return false
+
+    if (user.isSupport || user.isAdmin) return {}
+
+    if (user.type === STAFF) {
```
[the remark this answers is not in the document]
Comment (dkoviazin) [reply apps/condo/domains/billing/access/BillingReceiptFile.js:22]:
```
{
            OR: [
                {
                    organization: {
                        OR: [
                            queryOrganizationEmployeeFor(user.id),
                            queryOrganizationEmployeeFromRelatedOrganizationFor(user.id),
                        ],
                    },
                },
                { createdBy: { id: user.id } },
            ],
        }
    }
```

replace with

```
  {
      context: {
                    organization: {
                        OR: [
                            queryOrganizationEmployeeFor(user.id),
                            queryOrganizationEmployeeFromRelatedOrganizationFor(user.id),
                        ],
                    },
     }
}
```

there is no cases that employee will add billingReceiptFile - so  { createdBy: { id: user.id } } will not work
and as you do not have organization field in model you can query organization only through context

File: apps/condo/domains/billing/access/BillingReceiptFile.js [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
```
Comment (SavelevMatthew) [inline apps/condo/domains/billing/access/BillingReceiptFile.js:22]: Why?  

You've already have this branch in `canReadBillingEntity `

File: apps/condo/domains/billing/components/ServicesModal.tsx [hunk 10]
```diff
@@ -122,6 +124,27 @@ export const ServicesModal: React.FC<IServicesModalProps> = ({
...
+        where: {
+            receipt: { id: receipt && receipt.id },
+        },
+    })
+
+    const ModalFooter = () => {
+        return (
+            <Row justify='end'>
+                <Col span={24}>
+                    <Button loading={loading} onClick={() => window.open(get(objs[0], 'file.publicUrl'))} type='primary'>
```
Comment (SavelevMatthew) [inline apps/condo/domains/billing/components/ServicesModal.tsx:140]: `router.push`?

File: apps/condo/domains/billing/schema/BillingReceiptFile.js [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+        receipt: {
+            schemaDoc: 'Link to Billing Receipt',
+            type: Relationship,
+            ref: 'BillingReceipt',
+            isRequired: true,
+            knexOptions: { isNotNullable: true }, // Required relationship only!
+            kmigratorOptions: { null: false, on_delete: 'models.CASCADE' },
+        },
+
+        controlSum: {
```
Comment (SavelevMatthew) [inline apps/condo/domains/billing/schema/BillingReceiptFile.js:47]: Why not just JSON meta field like in other models?

File: apps/condo/domains/common/components/containers/GlobalStyle.tsx [hunk 12]
```diff
@@ -302,6 +302,12 @@ export default function GlobalStyle () {
               }
 
               .ant-table-cell {
+                .icon {
```
Comment (SavelevMatthew) [inline apps/condo/domains/common/components/containers/GlobalStyle.tsx:305]:
```
import { colors } from '@open-condo/ui/colors'
```

**Pull request merged**